examples/auth/certificate/certificate.go

```
@@ -150,75 +94,56 @@ func main() {
-	} else {
-		fmt.Printf("   ✅ SUCCESS: Server correctly denied access with status %d (Unauthorized)\n", expectedErrorCode)
+	clientWallet, err := exampleWallet.NewExampleWallet(exampleWallet.ExampleWalletArgs{
+		Type:       exampleWallet.ExampleWalletArgsTypePrivateKey,
+		PrivateKey: cPrivKey,
+	})
+	if err != nil {
+		panic(err)
 	}
 
-	fmt.Println("\n📡 STEP 3: Sending valid age verification certificate")
-	response2 := sendCertificate(mockedWallet, responseData.IdentityKey, responseData.InitialNonce)
-	if response2.StatusCode() != http.StatusOK {
-		fmt.Printf("   ❌ ERROR: Certificate submission failed with status: %d\n", response2.StatusCode())
-	} else {
-		fmt.Println("   ✅ SUCCESS: Server accepted the age verification certificate")
-	}
+	fmt.Println("✓ Client mockWallet created")
 
-	fmt.Println("\n📡 STEP 4: Testing access to protected resource WITH valid certificate")
-	resp = callPingEndpoint(mockedWallet, responseData)
-	if resp.StatusCode() != http.StatusOK {
-		fmt.Printf("   ❌ ERROR: Access denied with status: %d\n", resp.StatusCode())
-	} else {
-		fmt.Println("   ✅ SUCCESS: Server granted access to protected resource")
-		fmt.Println("   ↪ Received response: \"Pong!\"")
-	}
+	fmt.Println("\n📡 STEP 1: Sending non general request to /.well-known/auth endpoint")
+	responseData := callInitialRequest(clientWallet)
+	fmt.Println("✓ Auth completed")
 
-	fmt.Println("\n============================================================")
-	fmt.Println("🎉 DEMO COMPLETED SUCCESSFULLY")
-	fmt.Println("============================================================")
-}
+	fmt.Println("\n📡 STEP 2: Sending general request to test authorization")
+	callPingEndpoint(clientWallet, responseData)
+	fmt.Println("✓ General request completed")
 
-// ========== Handlers ==========
+	fmt.Println("\n📡 STEP 3: Sending certificates")
+	sendCertificate(context.Background(), clientWallet, responseData)
```

You ignored an error returned from `sendCertificate`.
- Since it is only an example, consider not returning it at all. (If printing of a result is enough)

---

it's handled with log.Fatalf right now